### PR TITLE
Added tests for all and any functions

### PR DESCRIPTION
**Pull request by jpsullivan**
Just getting started here. If there are any other test cases you can think of that I missed here, feel free to let me know.

### CODE REVIEW

File: src/functions/any.js [hunk 1]
```diff
@@ -26,27 +28,75 @@ export const any = wrap({
...
+                }
             }
         }
         return false;
     },
+    tests: process.env.NODE_ENV !== "development" ? undefined : {
+        "basicUsage": (hi) => {
+            const mixed = [true, true, false, false];
+            const result = hi.any(mixed);
+            hi.assertEqual(result, true);
```
Comment (pineapplemachine) [inline src/functions/any.js]: code examples as much as possible should show practical use cases to explain the reason for the function existing. this is a good test! but as a basic usage example how about something like

``` js
const strings = ["hello", "world", "how", "are", "you"];
hi.assert(hi.any(strings, i => i.startsWith("h")); // "hello" starts with "h"
hi.assertNot(hi.any(strings, i => i.startsWith("x")); // None start with "x"
```

**Comment by pineapplemachine** [comment]
The only change I'd request is the the usage example - otherwise this is good!

Not a blocker since more tests is infinity% better than no tests, but remember to test the weird cases - do the functions work when the input is an empty sequence? do the functions gracefully fail when the input is an unbounded sequence?

For example

```
"noInputs": hi => {
    hi.assertEqual(hi.emptySequence().any(), false);
},
"notKnownBoundedInput": hi => {
    hi.assertFail(
        (error) => (error.type === "NotBoundedError"),
        () => hi.recur(i => i + 1).seed(0).until(i => i === 100).any()
    );
},
"unboundedInput": hi => {
    hi.assertFail(
        (error) => (error.type === "NotBoundedError"),
        () => hi.repeatElement(0).any()
    );
},
```

File: src/functions/any.js [hunk 2]
```diff
@@ -26,27 +28,75 @@ export const any = wrap({
...
+        },
+        "sequenceWithPredicate": (hi) => {
+            const evenAndOdd = [1, 3, 5, 7, 9, 10, 11];
+            const result = hi(evenAndOdd).any((n) => n % 2 === 0);
+            hi.assertEqual(result, true);
+        },
+        "predicateAndSourceReversed": (hi) => {
+            const evenAndOdd = [1, 3, 5, 7, 9, 10, 11];
+            const result = hi.any(evenAndOdd, (n) => n % 2 === 0);
+            hi.assertEqual(result, true);
```
Comment (pineapplemachine) [inline src/functions/any.js:92]: An automatic tester should be responsible for making sure `hi(x).any()` and `hi.any(x)` behave the same - probably best to only do one ourselves

**Comment by pineapplemachine** [comment]
why are review comments not posted as soon as I submit them whyyyyyy??
